azavea/cac-tripplanner
# Fix mobile tour swipe index sync

## Overview

On mobile, when swiping through the locations of a tour or multi-location event in map view, the location highlighted on the map did not match the location on the active card. This was observed on iOS Safari, Chrome Android, and Chrome desktop at mobile widths, but was not universally reproducible.

It turns out that tiny-slider uses a one-based index while jquery uses a zero-based index. Further, tours and multi-location events have an initial slide for the tour/event description, which doesn't map to a specific location. This PR takes those into account when calculating the index of the location to highlight on the map.

While working on this, I observed that the itinerary carousel when getting point-to-point directions was broken by DOM changes in #1194. This PR fixes that and also addresses the above tiny-slider vs jquery index mismatch for itineraries.


## Testing Instructions

* Mobile
  * Open a tour map. Swipe through all locations. Ensure highlighted map marker and route match the selected card.
  * Open a multi-location event map. Swipe through all locations. Ensure highlighted map marker matches the selected card.
  * Open an itinerary map. Swipe through all itineraries. Ensure highlighted route matches the selected card.
* Desktop
  * Make sure map sidebars for tours, multi-location events, and itineraries all look and behave as expected.

Connects #1191
